src/main/java/com/intuit/graphql/orchestrator/batch/GraphQLServiceBatchLoader.java

```diff
@@ -46,8 +44,13 @@
 import java.util.stream.Collectors;
 import org.apache.commons.lang3.StringUtils;
 import org.dataloader.BatchLoader;
+import org.apache.commons.collections4.CollectionUtils;
+import com.intuit.graphql.orchestrator.authorization.DeclinedField;
+import com.intuit.graphql.orchestrator.authorization.FieldAuthorization;
+import com.intuit.graphql.orchestrator.authorization.QueryRedactor;
+import graphql.analysis.QueryTransformer;
 
-public class GraphQLServiceBatchLoader implements BatchLoader<DataFetchingEnvironment, DataFetcherResult<Object>> {
+public class GraphQLServiceBatchLoader<AuthDataT> implements BatchLoader<DataFetchingEnvironment, DataFetcherResult<Object>> {
```

Comment:
I dont think this class needs to be generic. Does it give a compilation error?

Reply:
Yes it gives compile error without the Generic Type.